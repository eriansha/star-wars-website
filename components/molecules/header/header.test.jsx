import { cleanup, render } from '@testing-library/react'
import ReactDOM from 'react-dom'

import Header from './header'

afterEach(cleanup)

describe('Footer', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(<Header />, div)
  })

  // TODO: fix unit test
  // it('contains image', () => {
  //   const alt='starwars banner'
  //   const src = '/img/brand.png'

  //   const tree = render(<Header />)
  //   const img = tree.getByAltText(alt)
  //   expect(img.getAttribute('src')).toBe(src);
  // })

  it('matches snapshot', () => {
    const tree = render(<Header />)
    expect(tree.container).toMatchSnapshot();
  })
})
