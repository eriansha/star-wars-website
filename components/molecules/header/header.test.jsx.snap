// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer matches snapshot 1`] = `
<div>
  <section
    class="bg-gray-dark border-b-2 border-gray flex gap-5 items-center justify-content-start pl-5 py-5"
  >
    <a
      href="/"
    >
      <span
        style="box-sizing: border-box; display: inline-block; overflow: hidden; width: 75px; height: 50px; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
      >
        <img
          alt="starwars banner"
          data-nimg="fixed"
          decoding="async"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
        />
        <noscript />
      </span>
    </a>
    <a
      href="/films"
    >
      Movie
    </a>
    <a
      href="/people"
    >
      People
    </a>
    <a
      href="/planet"
    >
      Planet
    </a>
    <a
      href="/species"
    >
      Species
    </a>
    <a
      href="/starship"
    >
      Starship
    </a>
  </section>
</div>
`;
